tests/models/shadow/model.p

```diff
@@ -0,0 +1,45 @@
+module System = {
+  Storage,
+  Partition,
+  Broker,
+  Producer,
+  CommitProtocol,
+  FetchProtocol
+};
+
+machine MultiPartitionProduceTest {
+  start state Init {
+    entry {
+      var storage: Storage;
+      var broker: Broker;
+      var producer: Producer;
+      var partitions: set[Partition];
+      var num_partitions: int;
+      var num_batches: int;
+
+      num_partitions = 3;
+      num_batches = 10;
+
+      storage = new Storage();
+
+      while (sizeof(partitions) < 3) {
+        partitions += (new Partition());
+      }
+
+      broker = new Broker((storage = storage,));
+
+      producer = new Producer((
+        broker = broker,
+        partitions = partitions,
+        count = num_batches));
+    }
+  }
+}
+
+test tcMultiPartitionProduce [main=MultiPartitionProduceTest]:
+  assert ProduceRequestResponse in
+  (union System, { MultiPartitionProduceTest });
+
+test tcFindConfiguration [main=MultiPartitionProduceTest]:
```

Comment:
This reliably fails if I run it with `time p compile && p check -tc tcFindConfiguration --schedules 1000`

```
<ErrorLog> PImplementation.SelectStorageConfiguration detected liveness bug in hot state 'TargetConfig' at the end of program execution.
```

I don't understand exactly what you are trying to test for there. Can you explain in english before I try to comment on it?

Reply:
> Are you running p check to look for a failing schedule and then you use the same schedule while modifying some other parts of the code?

No, it's even much simpler than that. The only thing I use it for is to make sure that the model is exploring interesting schedules. I mean, I think I should be exploring all permutations of L0d objects, but am I? So I just grab choose a few at random and see if the schedule finds them. I suspect I'll remove this monitor before too long as I discover better ways to do the same thing (new P user).